prediction_market_agent/agents/autogen_general_agent/farcaster_agent.py

```diff
@@ -0,0 +1,155 @@
+    recipient: UserProxyAgent,
+    messages: list[Dict[Any, Any]],
+    sender: AssistantAgent,
+    config: Optional[Any] = None,
+) -> str:
+    reflect_prompt = Template(CRITIC_PROMPT).substitute(
+        TWEET=recipient.chat_messages_for_summary(sender)[-1]["content"]
+    )
+    return reflect_prompt
+
+
+def build_llm_config(model: str) -> Dict[str, Any]:
+    keys = APIKeys()
+    return {
+        "config_list": [
+            {"model": model, "api_key": keys.openai_api_key.get_secret_value()}
+        ],
+    }
+
+
+def build_agents(model: str) -> Dict[AutogenAgentType, autogen.ConversableAgent]:
+    llm_config = build_llm_config(model)
+
+    user_proxy = autogen.UserProxyAgent(
+        name="User",
+        human_input_mode="NEVER",
+        is_termination_msg=lambda x: x.get("content", "").find("TERMINATE") >= 0,
+        code_execution_config={
+            "last_n_messages": 1,
+            "work_dir": "tasks",
+            "use_docker": False,
+        },
+    )
+
+    writer = autogen.AssistantAgent(
+        name="Writer",
+        llm_config=llm_config,
+        system_message="""
+        You are a professional writer, known for your insightful and engaging articles.
+        You transform complex concepts into compelling narratives.
+        You should imporve the quality of the content based on the feedback from the user.
+        """,
+    )
+
+    critic = autogen.AssistantAgent(
+        name="Critic",
+        llm_config=llm_config,
+        system_message="""
+            You are a critic, known for your thoroughness and commitment to standards.
+            Your task is to scrutinize content for any harmful elements or regulatory violations, ensuring
+            all materials align with required guidelines.
+            For code
+            """,
+    )
+
+    return {
+        AutogenAgentType.CRITIC: critic,
+        AutogenAgentType.WRITER: writer,
+        AutogenAgentType.USER: user_proxy,
+    }
+
+
+def build_tweet(
+    model: str,
+) -> str:
+    """
+    Builds a tweet based on the five markets on Omen that are closing soonest.
+
+    This function utilizes the writer and critic agents to generate the tweet content. It first initializes the
+    necessary agents based on the provided model. Then, it registers a chat between the writer and critic agents.
+
+    The tweet content is generated using a template that includes questions about each market's title and likelihood.
+    """
+
+    agents = build_agents(model)
+    user_proxy, writer, critic = (
+        agents[AutogenAgentType.USER],
+        agents[AutogenAgentType.WRITER],
+        agents[AutogenAgentType.CRITIC],
+    )
+    user_proxy.register_nested_chats(
+        [
+            {
+                "recipient": critic,
+                "message": reflection_message,
+                "summary_method": "last_msg",
+                "max_turns": 1,
+            }
+        ],
+        trigger=writer,  # condition=my_condition,
```

Comment:
What's the condition for?

Reply:
removed, it was part of Autogen docs. Idea is that you can add a condition for the nested chat to occur.